tests/queries/0_stateless/03377_bloom_filter_skip_index.sql

```diff
@@ -0,0 +1,56 @@
+--- Checks if the skip indices are correctly used depending on the filter predicates.
+
+DROP TABLE IF EXISTS `example`;
+CREATE TABLE `example`
+(
+    `id` UInt32,
+    `colA` String,
+    `colB` String,
+    INDEX `colA_idx` colA TYPE bloom_filter GRANULARITY 1,
+    INDEX `colB_idx` colB TYPE bloom_filter GRANULARITY 1
+)
+ENGINE = MergeTree
+ORDER BY `id`
+SETTINGS index_granularity = 1;
+
```

Comment:
Asking myself if other skip indexes are affected by the same problem as well? See [here](https://clickhouse.com/docs/engines/table-engines/mergetree-family/mergetree#table_engine-mergetree-data_skipping-indexes) for a list of skipping indices.

Reply:
Yes, I tried `minmax` and `set` skip index types and they are struggling with the same issue. I guess we can unify the implementation to use the new one, but this might require some rework.

Reply:
If minmax and set indexes are also affected, then we should definitely also fix them. So, yes, a refactoring would be appreciated. Ideally, it is pushed as a separate PR first, and then we could push a small fix PR on top. That way, we wouldn't need to roll everything in case some tests fail later on.

(It is difficult for me to review this PR this week due to time reasons. I think I can check next week in more detail.)